src/wallet/wallet.cpp

```diff
@@ -4071,16 +4010,69 @@ bool CWallet::UpgradeWallet(int version, bilingual_str& error, std::vector<bilin
     return true;
 }
 
-void CWallet::postInitProcess()
+CWallet::ScanStatus CWallet::HandleNotifications(std::shared_ptr<CWallet> wallet, bool scan)
 {
-    LOCK(cs_wallet);
+    // Register with the validation interface. Skip requesting mempool transactions if wallet is empty.
+    interfaces::Chain::ScanFn scan_fn;
+    interfaces::Chain::MempoolFn mempool_fn;
+    Optional<CBlockLocator> best_block_locator;
+    // The way the 'time_first_key' is initialized is just a workaround for the gcc bug #47679 since version 4.6.0.
+    Optional<int64_t> time_first_key = MakeOptional(false, int64_t());;
+    ScanStatus scan_status = ScanStatus::SKIPPED;
+    if (scan) {
+        // Get best block locator to rescan if not going back and rescanning
+        if (!gArgs.GetBoolArg("-rescan", false)) {
+            best_block_locator.emplace();
+            WalletBatch(*wallet->database).ReadBestBlock(*best_block_locator);
+        }
 
+        // No need to read and scan block if block was created before
+        // our wallet birthday (as adjusted for block time variability)
```

Comment:
Block time variability is unclear, you mean the absence of order guarantee between `nTime`  and `nTimeFirstKey`?